### ISSUE TITLE
All services except FingerID RestAPI available (Proxy)

### ISSUE THREAD

**Issue by SteffenHeu** [boecker-lab/sirius#68]
Hey, 

I have to use a proxy from my Uni PC. I set the proxy settings in the configuration (either manually from Sirius or automatically via the system option).

It seems to work for everything except the finger ID rest API (see screenshot below).

![grafik](https://user-images.githubusercontent.com/37407705/162969971-75769855-e0b1-437b-8f96-7a5df4549476.png)

Anything i did wrong or i can try to fix the issue?

Thanks in advance

Steffen

**Comment by mfleisch** [comment]
Hey Steffen,
can you provide a log output? Either by copying from the log window or by running the gui from the command line  (`sirius gui`) and copying the command line output.

Educated guess:
We observed that with some proxies the ssl validation fails after the traffic has been routed through the proxy. We have introduced several improvements regarding proxy support in SIRIUS 5 including the possibility to disable ssl validation.

So it might be worth trying the pre-release of SIRIUS 5 with disabled ssl validation to check whether this solves the issue.
https://github.com/boecker-lab/sirius/releases/tag/v5.4.1-SNAPSHOT

In general, I would highly recommend to use the SIRIUS proxy settings. Support for system proxies will be dropped with SIRIUS 5 anyways since they never worked reliably.

**Comment by SteffenHeu** [comment]
Hi Markus,

here's the log, i don't know if it helps:  (Set to ALL, running from the console just displayed level INFO)

```
Apr. 12, 2022 7:23:14 PM de.unijena.bioinf.ms.frontend.subtools.CLIRootOptions configureProjectSpace
WARNUNG: No unique output location found. Writing output to Temporary folder: C:\Users\a\AppData\Local\Temp\.sirius-tmp-project-8812039959912528938
Apr. 12, 2022 7:23:14 PM de.unijena.bioinf.ms.frontend.subtools.CLIRootOptions configureProjectSpace
WARNUNG: No unique output location found. Writing output to Temporary folder: C:\Users\a\AppData\Local\Temp\.sirius-tmp-project-8812039959912528938
Apr. 12, 2022 7:23:14 PM de.unijena.bioinf.ms.frontend.subtools.CLIRootOptions configureProjectSpace
WARNUNG: No unique output location found. Writing output to Temporary folder: C:\Users\a\AppData\Local\Temp\.sirius-tmp-project-8812039959912528938
Apr. 12, 2022 7:23:14 PM de.unijena.bioinf.jjobs.JJob lambda$logInfo$9
INFORMATION: <1>[JJob] No Input has been imported to Project-Space. Starting application without input data.
Apr. 12, 2022 7:23:14 PM de.unijena.bioinf.jjobs.JJob lambda$logInfo$9
INFORMATION: <2>[JJob] Painting GUI...
Apr. 12, 2022 7:23:14 PM de.unijena.bioinf.jjobs.JJob lambda$logInfo$9
INFORMATION: <2>[JJob] Checking Webservice connection...
Apr. 12, 2022 7:24:40 PM de.unijena.bioinf.jjobs.JJob lambda$logDebug$13
FEIN: <10>[CheckJob] Running post processing
Apr. 12, 2022 7:24:40 PM de.unijena.bioinf.jjobs.JJob lambda$logDebug$13
FEIN: <10>[CheckJob] Running cleanup
Apr. 12, 2022 7:24:40 PM de.unijena.bioinf.jjobs.JJob lambda$logDebug$13
FEIN: <9>[JJob] Running post processing
Apr. 12, 2022 7:24:40 PM de.unijena.bioinf.jjobs.JJob lambda$logDebug$13
FEIN: <9>[JJob] Running cleanup
```

Something that just came to my mind: When fixing the pubchem lookup in MZmine 3 for proxies i noticed that the API internally resolved via https and we just set a http proxy. Setting the same proxy to https solved the problem. However, neither selecting http or https solved the issue in Sirius 4, so you are probably right with your guess.

It works fine with Sirius 5 however.

Is there a changelog i can look at to see the differences between Sirius 4 and 5? 

I'm working on a simple MZmine 3 integration, so it would be good to see if there is something to be aware of.

**Comment by mfleisch** [comment]
I think building the MZmine 3 integration for SIRIUS 5 might be a good idea anyways since there are quite some breaking changes compared to SIRIUS 4 and SIRIUS 4 will reach its end of life in the foreseeable future.

I added the most critical changes to the [changelog](https://boecker-lab.github.io/docs.sirius.github.io/changelog/#541). However it is not finished yet. 

A guess you are working on an integration based on the sirius libraries which are unfortunately still poorly documented.
So, let me know if I can help with some information regarding entry points or code stubs.

**Comment by SteffenHeu** [comment]
Actually, i was trying to make it work via the CLI interface. Since the were problems keeping the old finger ID web api in MZmine 2 up to date, i thought it might be the better solution.

Creating custom databases and running formula and structure works fine so far. And i also made a simple import for the sirius results from the project space. Reading the changelogs, i guess that one needs updating for Sirius 5.

Looking at the repository for the sirius lib https://github.com/boecker-lab/sirius-libs it looks like it is for an older version of Sirius 4, so i guess i'm better off using the current CLI method, or?

**Comment by mfleisch** [comment]
Using the CLI is totally fine. I just remembered that the libs were used for MZmine 2 und thought it might be the same for MZmine 3.

Using the CLI definitely has the advantage that the user can update SIRIUS, without you having to update MZmine all the time. We attach great importance to backward compatibility regarding project-space and CLI unless there ist not a major version bump of SIRIUS.

Adapting to the project-space changes should be straight forward when using java.nio.ZipFs  to open the archives that have been directories before. 
The `fingerprint`, `trees`,  etc.  directories are now zip archives. But the content inside the archives does not have really changed compared to the dirs. So parsers should still work.

If you are using the summaries tsv files I would recommend using parsers that rely on the column header and  not  on the  column index. Because we will not treat adding columns to these tsv files as breaking change.


The user authentication is token based and stores a refresh token on the system when a user has logged in. 
So you do not have to care about authentication if the user has logged in using SIRIUS once. You could also execute the login from MZmine via the SIRIUS cli without having to store any user credentials.

**Comment by mfleisch** [comment]
I  close this issue since the proxy problem seems to be solved in SIRIUS v5.4.
For further discussion on MZmine integration we should create a different issue.
